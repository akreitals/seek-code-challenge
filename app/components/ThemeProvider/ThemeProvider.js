import React, { Fragment } from 'react';
import { ThemeProvider as Root, injectGlobal } from 'styled-components';
import PropTypes from 'prop-types';
import themeConfig, { colors } from '../theme';

injectGlobal([
    `
    html {
        box-sizing: border-box;
        text-rendering: optimizeLegibility;
        -webkit-font-smoothing: antialiased;
        -webkit-appearance: none;
        -moz-appearance: none;
    }
    
    *,
    *::before,
    *::after {
        box-sizing: inherit;
    }
    
    html,
    body {
        min-height: 100%;
        min-width: 100%;
    }
    
    body {
        padding: 0;
        margin: 0;
        color: ${colors.black};
        font-size: ${themeConfig.fontSizes[2]}px;
        font-family: ${themeConfig.font};
        line-height: 1.5;
        position: relative;
        height: 100%;
        max-height: 100%;
        width: 100%;
        -webkit-font-smoothing: antialiased;
        overflow-x: hidden;
        -webkit-overflow-scrolling: touch;
    }
    
    a {
        color: ${colors.blueLighter};
        text-decoration: none;
    }
    
    textarea {
        resize: none;
    }
`
]);

/* eslint-disable react/no-children-prop */
const ThemeProvider = ({ theme, ...props }) => (
    <Root
        theme={themeConfig}
        {...props}
        children={<Fragment>{props.children}</Fragment>}
    />
);
/* eslint-enable */

ThemeProvider.propTypes = {
    theme: PropTypes.object
};

ThemeProvider.defaultProps = {
    themeConfig
};

export default ThemeProvider;
