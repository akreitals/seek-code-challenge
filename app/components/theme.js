const maxContainerWidth = '940px';

const emoji = '"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"';
const font = `Averta, -apple-system, BlinkMacSystemFont, “Segoe UI”, Roboto, Helvetica, Arial, sans-serif, ${emoji}`;
const mono = 'SFMono-Regular,"Roboto Mono",Menlo,monospace';

const fontSizes = [12, 14, 16, 18, 24, 32, 48, 64, 72, 96];
const space = [0, 4, 8, 16, 32, 64, 128];
const regular = 400;
const bold = 600;

const greys = {
    black: '#1c1c1c',
    charcoal: '#404040',
    midGrayDark: '#747474',
    midGrayMedium: '#898989',
    midGray: '#a8a8a8',
    midGrayLight: '#d6d6d6',
    grayMedium: '#e1e1e1',
    grayLight: '#eeeeee',
    grayLightest: '#f7f7f7',
    offWhite: '#fbfbfb',
    white: '#ffffff'
};

const brand = {
    blueDarker: '#071d40',
    blueDark: '#042763',
    blue: '#0d3880',
    blueLight: '#184da8',
    blueLighter: '#2765cf',
    pink: '#e60278',
    green: '#157e00',
    greenLight: '#5bc252',
    purple: '#9556b7',
    teal: '#00a5ad',
    orange: '#f57c00',
    yellow: '#ffc600',
    yellowLight: '#fffce3',

    primary: '#e60278',
    accent: '#9556b7',
    success: '#157e00',
    info: '#2765cf',
    warning: '#f57c00',
    error: '#e60278'
};

export const colors = {
    ...greys,
    ...brand
};

export const radius = '2px';

// boxShadows
export const boxShadows = [
    `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
    `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
];

const theme = {
    colors,
    font,
    mono,
    fontSizes,
    space,
    regular,
    bold,
    maxContainerWidth,
    radius,
    boxShadows
};

export default theme;
