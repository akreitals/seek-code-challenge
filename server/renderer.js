import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
