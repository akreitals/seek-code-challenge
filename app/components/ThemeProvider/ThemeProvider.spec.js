import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';
import { colors } from '../theme';
import ThemeProvider from './ThemeProvider';

const Content = styled.p`
    color: ${colors.primary};
`;

describe('ThemeProvider', () => {
    test('should render content with theme', () => {
        const json = renderer
            .create(
                <ThemeProvider>
                    <Content>Hello!</Content>
                </ThemeProvider>
            )
            .toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('color', colors.primary);
    });
});
