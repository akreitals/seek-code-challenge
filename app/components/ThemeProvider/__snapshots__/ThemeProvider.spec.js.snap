// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should render content with theme 1`] = `
.c0 {
  color: #e60278;
}

<p
  className="c0"
>
  Hello!
</p>
`;
